src/getDataset.js

```diff
@@ -18,6 +18,7 @@ const requestCertainFileType = async (res, req, additional, query) => {
   if (query.type === "tree") {
     res.json({tree: jsonData});
   }
+  // TODO why does this work without a return?
   res.json(jsonData);
```

Comment:
I didn't have time to dig into this, but if anyone has some insight onto why sometimes we don't have to return a Response object, I'd love to learn more.

Reply:
The short answer is that `res.json(jsonData)` on its own will send the data to the client --  see the [express docs](https://expressjs.com/en/5x/api.html#res.json).

Digging one level deeper, `requestMainDataset` is called by `getDataset`, which doesn't store the return value, so what `requestMainDataset` returns doesn't matter. (In this case the return value is `undefined`.)

Digging another layer deeper, we see that `getDataset` returns a variety of things depending on the code path taken -- `undefined`, [`ServerResponse`](https://nodejs.org/api/http.html#http_class_http_serverresponse), and maybe others. That's probably not ideal in terms of reusing that code in other servers etc.

Reply:
If you want to dig even deeper, `getDataset` is the callback [we use to handle GET requests to `"/charon/getDataset"`](https://github.com/nextstrain/nextstrain.org/blob/master/server.js#L92-L93). From [the express docs](https://expressjs.com/en/api.html#app.route) it's not clear if the return value of this callback matters -- the examples there don't explicitly return anything -- but perhaps it is?!?